### PR TITLE
Add callback to register packs after vanilla 

### PR DESCRIPTION
**Pull request by Leo40Git**
This allows overriding vanilla resources!

The old singular `EVENT` callback has been renamed to `BEFORE_VANILLA`, to be more descriptive (old name still works, of course!)

(I also took the liberty of updating everything and adding `gradlew.bat` for us Windows users)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,5 +1,5 @@
 plugins {
-    id 'fabric-loom' version '0.4-SNAPSHOT'
+    id 'fabric-loom' version '0.6-SNAPSHOT'
```
Comment (FoundationGames) [inline build.gradle:2]: If you are going to update loom, why not 0.8?

Comment (Leo40Git) [reply build.gradle:2]: I generally avoid the latest snapshot of Loom.

**Comment by Leo40Git** [comment]
Can I just ask why the Gradle wrapper scripts were being gitignore'd?

**Comment by Devan-Kerman** [comment]
cus I'm dumb

**Comment by Devan-Kerman** [comment]
lgtm, updating to 1.16.5 was unnecessary but everyone's probably moved on by now

**Pull request merged**